packages/integration-tests/utils/browserHelpers.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * Helper to wait for an XHR to finish
+ *
+ * @param {{ readyState: number }} xhr
+ * @param {() => unknown} cb
+ * @returns {*}  {unknown}
+ */
+export function waitForXHR(xhr: { readyState: number }, cb: () => unknown): unknown {
+  if (xhr.readyState === XMLHttpRequest.DONE) {
+    return cb();
+  }
+
+  setTimeout(function() {
+    waitForXHR(xhr, cb);
+  }, 1000 / 60);
```

Comment:
So this can go infinite? Can we stop this after X amount of iterations?

Reply:
Feels simpler to rely on the timeouts we defined on Playwright configuration, what do you think?

Reply:
I don't remember how this was used, but considering it is now unused code, I'd suggest we delete it and discuss implementation later (and removing this would affect the README too, since it talks about this file).